### PR TITLE
[#1477] fix(web): fix tree view sync and add table loading

### PR DESCRIPTION
**Pull request by ch3yne**
### What changes were proposed in this pull request?

Improve tree view synchronization and add table loading.

https://github.com/datastrato/gravitino/assets/17310559/4e84481c-389a-4686-aa5c-2b7ade70aaf5

### Why are the changes needed?

Fix: #1477 fix tree view sync
Fix: #1605 fix table data fill when error
Fix: #1606 add table fetch loading
Fix: #1619 dup of 1605
Fix: #1644 dup of 1605

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by jerryshao** [comment]
@ch3yne can you please record a simple video or paste some screenshots about your fix?

**Comment by ch3yne** [comment]
Optimize the tree view expand effect.

https://github.com/datastrato/gravitino/assets/17310559/e73f5613-848e-4220-8885-98354ea9697b

File: web/app/ui/metalakes/MetalakePath.js [hunk 1]
```diff
@@ -20,7 +20,6 @@ const Text = styled(Typography)(({ theme }) => ({
 
 const MetalakePath = props => {
   const { routeParams } = props
-
```
Comment (jerryshao) [inline web/app/ui/metalakes/MetalakePath.js:23]: I think this is an unnecessary change, can you please revert this line?

Comment (ch3yne) [reply web/app/ui/metalakes/MetalakePath.js]: Reverted.

**Review by jerryshao — APPROVED** [review]
